src/storage/v2/disk/label_property_index.cpp

```diff
@@ -0,0 +1,272 @@
+    if (auto tx_it = tx_to_entries_for_deletion.find(tx.start_timestamp); tx_it != tx_to_entries_for_deletion.end()) {
+      if (auto vertex_label_index_it = tx_it->second.find(vertex_after_update->gid);
+          vertex_label_index_it != tx_it->second.end()) {
+        std::erase_if(vertex_label_index_it->second, [added_label](const std::pair<LabelId, PropertyId> &index) {
+          return index.first == added_label;
+        });
+      }
+    }
+  });
+}
+
+void DiskLabelPropertyIndex::UpdateOnRemoveLabel(LabelId removed_label, Vertex *vertex_after_update,
+                                                 const Transaction &tx) {
+  for (const auto &index_entry : index_) {
+    if (index_entry.first == removed_label) {
+      entries_for_deletion.WithLock([&index_entry, &tx, vertex_after_update](auto &tx_to_entries_for_deletion) {
+        const auto &[indexing_label, indexing_property] = index_entry;
+        auto [it, _] = tx_to_entries_for_deletion.emplace(
+            std::piecewise_construct, std::forward_as_tuple(tx.start_timestamp), std::forward_as_tuple());
+        auto &vertex_map_store = it->second;
+        auto [it_vertex_map_store, emplaced] = vertex_map_store.emplace(
+            std::piecewise_construct, std::forward_as_tuple(vertex_after_update->gid), std::forward_as_tuple());
+        it_vertex_map_store->second.emplace_back(indexing_label, indexing_property);
+      });
+    }
+  }
+}
+
+/// TODO: andi If stays the same, move it to the hpp
+void DiskLabelPropertyIndex::UpdateOnSetProperty(PropertyId property, const PropertyValue &value, Vertex *vertex,
+                                                 const Transaction &tx) {}
+
+bool DiskLabelPropertyIndex::DropIndex(LabelId label, PropertyId property) {
+  return index_.erase({label, property}) > 0;
+}
+
+bool DiskLabelPropertyIndex::IndexExists(LabelId label, PropertyId property) const {
+  return utils::Contains(index_, std::make_pair(label, property));
+}
+
+std::vector<std::pair<LabelId, PropertyId>> DiskLabelPropertyIndex::ListIndices() const {
+  return std::vector<std::pair<LabelId, PropertyId>>(index_.begin(), index_.end());
+}
+
+// These constants represent the smallest possible value of each type that is
+// contained in a `PropertyValue`. Note that numbers (integers and doubles) are
+// treated as the same "type" in `PropertyValue`.
+const PropertyValue kSmallestBool = PropertyValue(false);
+static_assert(-std::numeric_limits<double>::infinity() < std::numeric_limits<int64_t>::min());
+const PropertyValue kSmallestNumber = PropertyValue(-std::numeric_limits<double>::infinity());
+const PropertyValue kSmallestString = PropertyValue("");
+const PropertyValue kSmallestList = PropertyValue(std::vector<PropertyValue>());
+const PropertyValue kSmallestMap = PropertyValue(std::map<std::string, PropertyValue>());
+const PropertyValue kSmallestTemporalData =
+    PropertyValue(TemporalData{static_cast<TemporalType>(0), std::numeric_limits<int64_t>::min()});
```

Comment:
I would change this from copy ctor to regular ctor using braced init

```suggestion
const PropertyValue kSmallestNumber{-std::numeric_limits<double>::infinity()};
const PropertyValue kSmallestString{""};
const PropertyValue kSmallestList{std::vector<PropertyValue>()};
const PropertyValue kSmallestMap{std::map<std::string, PropertyValue>()};
const PropertyValue kSmallestTemporalData{TemporalData{static_cast<TemporalType>(0), std::numeric_limits<int64_t>::min()}};
```